Support netrc-based credentials in buf-setup-action (#94)
bufbuild/buf-setup-action: README.md

|:---------------|:---------------------------------------------------|:-------------------|
| `version`      | The version of the [`buf` CLI][buf-cli] to install | [`v1.11.0`][version] |
| `github_token` | The GitHub token to use when making API requests   |                    |

> | `buf_user`     | The The username to use for logging into Buf Schema registry.                                               |                    |

Reviewer: ```suggestion
| `buf_user`     | The username to use for logging into Buf Schema registry.                                               |                    |
```

      github_token: ${{ github.token }}
```

> ### Buf username and Buf API token

Reviewer: ```suggestion
### Authenticating with the Buf Schema Registry
```

Reviewer: maybe also worth putting a link to the docs site "for more information" here unless its already mentioned

Author: This part is quite confusing since it mentions parameters and other configurations in the configurations in the same section. I think it might be a better idea to separate the parameters and other configurations.

Reviewer: I also see another problem, we speak about authenticating with the BSR using 2 different methods: on with a username and api token, the other with just a token. Is there a way we might be able to distinguish between the two usecases?
1. The .netrc file allows users to quickly login to the BSR system-wide without manually having to reenter the credentials for each tool(includes private go modules).  
2. The token env var authenticates just `Buf`, not the whole system.

Author: I made the changes:
1. I mentioned the access to private remote packages in the section of supplying username and buf api token
2. I also mentioned in the Buf Token part that specifying BUF_TOKEN as env var does does gain the access to private remote packages. 

I will bring this up briefly in the doc site.

      github_token: ${{ github.token }}
```

> #### Buf username and Buf API token
> 
> Optionally, you can supply both `buf_user` and `buf_api_token` input so that it will perform 
> [Buf Schema Registry][bsr] (BSR) authentication at the same time. This will allow you to access 
> the private remote packages in BSR. This only works when both `buf_user` and `buf_api_token` are
> supplied at the same time:
> 
> ```yaml
> steps:
>   - uses: bufbuild/buf-setup-action@v1.11.0
>     with:
>       buf_user: ${{ secrets.buf_user }}
>       buf_api_token: ${{ secrets.buf_api_token }}
> ```

Reviewer: ```suggestion
#### Buf username and Buf API token

If you are using Private [Remote Packages](https://docs.buf.build/bsr/remote-packages/overview) you may need 
to authenticate the entire system to successfully communicate with the [Buf Schema Registry][bsr]. To achieve this, supply both `buf_user` and `buf_api_token`. This will add your auth credentials to the `.netrc` and allow you to access the BSR from anything in your `PATH`. 

```yaml
steps:
  - uses: bufbuild/buf-setup-action@v1.11.0
    with:
      buf_user: ${{ secrets.buf_user }}
      buf_api_token: ${{ secrets.buf_api_token }}
```
```

> ```

Reviewer: so the two methods to auth with buf are separated?

> ```

Author: **buf_token** is a env var instead of an input to buf-setup-action. That's why I want to separate them by "Input" and "Other Configuration". I just changed the section name from "parameter" to "input".
